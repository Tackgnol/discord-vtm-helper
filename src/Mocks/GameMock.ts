import { Game } from '../Models/AppModels';
import { activeSessionMock } from './SessionDataMocks/ActiveSessionMock';
import { playerMock } from './SessionDataMocks/PlayerMock';

export const gameMock: Game = {
	activeChannel: activeSessionMock.channelId,
	adminId: '0',
	channels: [activeSessionMock],
	current: true,
	id: 'x',
	players: [playerMock],
};
