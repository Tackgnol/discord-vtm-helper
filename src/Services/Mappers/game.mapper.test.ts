import { gameMapper } from './game.mapper';
import { gameMock } from '../../Mocks/GameMock';
import chai, { expect } from 'chai';
import { jestSnapshotPlugin } from 'mocha-chai-jest-snapshot';

chai.use(jestSnapshotPlugin());

describe('Services >> Mappers > gameMapper', () => {
	it('maps a game on proper data input', () => {
		const result = gameMapper(gameMock);
		expect(result).toMatchSnapshot();
		expect(result.channels).to.have.length.above(0);
		expect(result.players).to.have.length.above(0);
	});

	it('throws an error on missing gameId', () => {
		const faultyGameMock = { ...gameMock, id: undefined };
		expect(() => {
			gameMapper(faultyGameMock);
		}).to.throw(Error);
	});

	it('throws an error on missing adminId', () => {
		const faultyGameMock = { ...gameMock, adminId: undefined };
		expect(() => {
			gameMapper(faultyGameMock);
		}).to.throw(Error);
	});

	it('throws an error on missing current', () => {
		const faultyGameMock = { ...gameMock, current: undefined };
		expect(() => {
			gameMapper(faultyGameMock);
		}).to.throw(Error);
	});
	it('throws an error on missing active channel', () => {
		const faultyGameMock = { ...gameMock, activeChannel: undefined };
		expect(() => {
			gameMapper(faultyGameMock);
		}).to.throw(Error);
	});

	it('throws an error on missing channels', () => {
		const faultyGameMock = { ...gameMock, channels: undefined };
		expect(() => {
			gameMapper(faultyGameMock);
		}).to.throw(Error);
	});

	it('throws an error on missing players', () => {
		const faultyGameMock = { ...gameMock, players: undefined };
		expect(() => {
			gameMapper(faultyGameMock);
		}).to.throw(Error);
	});
});
