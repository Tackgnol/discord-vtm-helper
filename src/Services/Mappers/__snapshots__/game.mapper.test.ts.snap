// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Services >> Mappers > gameMapper maps a game on proper data input 1`] = `
Object {
  "activeChannel": "0",
  "adminId": "0",
  "channels": Array [
    Object {
      "channelId": "0",
      "globaltestSet": undefined,
      "multiMessageSet": undefined,
      "narrationSet": undefined,
      "statInsightSet": undefined,
    },
  ],
  "current": true,
  "id": "x",
  "players": Array [
    Object {
      "discordUserName": "test",
      "id": "0",
      "npcSet": Array [
        Object {
          "callName": "testWithFacts",
          "description": "this is a test",
          "facts": Array [
            "fact1",
            "fact2",
          ],
          "gameId": "x",
          "image": "placeholder",
          "name": "Test",
        },
        Object {
          "callName": "testWithout",
          "description": "this is a test",
          "facts": Array [],
          "gameId": "x",
          "image": "placeholder",
          "name": "Test",
        },
      ],
      "statisticsSet": Array [
        Object {
          "name": "test",
          "value": 2,
        },
      ],
    },
  ],
}
`;
